scripts/issue/71/consolidate_badger.py

```diff
@@ -0,0 +1,145 @@
+        # Beneficiary is currently the DAO_Agent
+        assert dao_treasury.beneficiary() == registry.eth.badger_wallets.fees
+
+        # Nonces are messed up on this multi - next in line is 78
+        old_multi.post_safe_tx(replace_nonce=78)
+
+    else:
+        # Simulates upgrade of contract
+        if simulation == "true":
+            new_logic = registry.eth.logic.SimpleTimelockWithVoting
+            old_devProxyAdmin.upgrade(dao_treasury.address, new_logic)
+
+        safe = GreatApeSafe(DEV_MULTISIG)
+        treasury_vault = GreatApeSafe(TREASURY_VAULT)
+        safe.take_snapshot(tokens=[registry.eth.treasury_tokens.BADGER])
+        treasury_vault.take_snapshot(tokens=[registry.eth.treasury_tokens.BADGER])
+
+        # Contracts needed
+        controller = interface.IController(
+            registry.eth.controllers.native,
+            owner=safe.account
+        )
+        balance_checker = interface.IBalanceChecker(
+            registry.eth.helpers.balance_checker,
+            owner=safe.account
+        )
+        dao_treasury = interface.ISimpleTimelockWithVoting(
+            registry.eth.badger_wallets.DAO_treasury,
+            owner=safe.account
+        )
+        bBadger_vault = interface.ISettV4h(
+            registry.eth.sett_vaults.bBADGER,
+            owner=safe.account
+        )
+        bBadger_strat = interface.IStrategy(
+            registry.eth.strategies["native.badger"],
+            owner=safe.account
+        )
+
+        ## == Release all BADGER from DAO_Treasury to devMulti == ##
+        
+        # Sets new beneficiary
+        dao_treasury.setBeneficiary(DEV_MULTISIG)
+        assert dao_treasury.beneficiary() == DEV_MULTISIG
+
+        treasury_balance = badger_token.balanceOf(dao_treasury.address)
+        gov_balance = badger_token.balanceOf(DEV_MULTISIG)
+
+        # Release BADGER to devMultisig
+        dao_treasury.release()
+
+        balance_checker.verifyBalance(
+            badger_token.address,
+            DEV_MULTISIG,
+            treasury_balance + gov_balance
+        )
+
+        ## == Transfer missing BADGER to Geyser and withdrawAll to vault == ##
+        geyser_balance = badger_token.balanceOf(BADGER_GEYSER)
+        strategy_balanceOf = bBadger_strat.balanceOf()
+        geyser_deficit = strategy_balanceOf - geyser_balance
+
+        # Ensure geyser is in deficit
+        assert geyser_deficit > 0
+        
+        # Transfer missing BADGER to Geyser
+        badger_token.transfer(BADGER_GEYSER, geyser_deficit, {"from": safe.address})
+
+        CONSOLE.print(f"{geyser_deficit / 1e18} BADGERs were transfer to the Geyser!\n")
+
+        balance_checker.verifyBalance(
+            badger_token.address,
+            BADGER_GEYSER,
+            geyser_deficit
+        )
+
+        # Once the Geyser has enough BADGER within, we can
+        # withdrawAll to transfer all BADGER from strat to the vault
+        assert bBadger_vault.available() == 0
```

Comment:
This line in case that there is idle badger in the sett will revert

ref: https://etherscan.io/address/0xe4ae305b08434bf3d74e0086592627f913a258a9#code#L1254

for example now it is greater than 0, there is 668 badger

<img width="1405" alt="Screenshot 2022-02-08 at 14 16 07" src="https://user-images.githubusercontent.com/84875062/152994524-c25f4c42-739f-40b4-b2e6-be428232d223.png">

Reply:
You are right, specially if we turn earn() off, BADGER could start accumulating. Will remove the assertion.